db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SQLGrammarImpl.java

```diff
@@ -724,6 +725,55 @@ StatementNode getUpdateNodeWithSub(FromTable fromTable, /* table to be updated *
         return retval;
     }
 
+    SubqueryNode assembleUpdateSubquery(ResultColumnList setClause, FromList fromList) throws StandardException {
+        ResultColumnList innerRCL = (ResultColumnList) nodeFactory.getNode(
+                C_NodeTypes.RESULT_COLUMN_LIST,
+                getContextManager());
+
+        boolean generateName;
+        for (int index = 0; index < setClause.size(); index++) {
+            ResultColumn rc = setClause.elementAt(index);
+            String columnName = rc.getExpression().getColumnName();
+            generateName = false;
+            if (columnName == null) {
+                columnName = "UPD_SUBQ_COL_" + index;
+                generateName = true;
+            }
+            ResultColumn innerRC = (ResultColumn) nodeFactory.getNode(
+                    C_NodeTypes.RESULT_COLUMN,
+                    columnName,
+                    rc.getExpression(),
+                    getContextManager());
+            if (generateName) {
+                innerRC.setNameGenerated(true);
+            }
+
+            innerRCL.addResultColumn(innerRC);
+        }
+
+        SelectNode selectNode = (SelectNode) nodeFactory.getNode(
+                C_NodeTypes.SELECT_NODE,
+                innerRCL,  /* SELECT list */
+                null, /* AGGREGATE list */
+                fromList,  /* FROM list */
+                null, /* WHERE clause */
+                null, /* GROUP BY list */
+                null, /* having clause */
+                null, /* window list */
+                getContextManager());
+
+        return (SubqueryNode) nodeFactory.getNode(
```

Comment:
Does it make sense to also have aggregations supported in the select subquery?

Reply:
Aggregation works. You can write a subquery that selects aggregations and put it in the from list. The `SelectNode` built here is merely a wrapper used in query processing. Suppose the user subquery has `select avg(x), sum(y) from ...`, the select we build here is something like
`select a, b from (select avg(x) as a, sum(y) as b from ...) ...`
That's why we don't have aggregation list or any other clauses in this select.

Of course, something like `update set x = sum(y) from a, b where ...` is beyond the scope, I think. It might be possible, though.